File: 05/220506/42577pro/42577.js
/* 전화번호 목록 */

const solution = (phoneBook) => {
  phoneBook.sort();
  for (let i = 0; i < phoneBook.length - 1; i++) {
    if (phoneBook[i + 1].startsWith(phoneBook[i])) {
      return false;
    }
  }
  return true;
};

const phoneBook = ['119', '97674223', '1195524421'];

console.log(solution(phoneBook));

File: 05/220506/42577pro/42577.py
# 전화번호 목록 

def solution(phone_book):
    phone_book.sort()
    for prev, cur in zip(phone_book, phone_book[1:]):
        if (cur.startswith(prev)):
            return False
    return True

phone_book = ["119", "97674223", "1195524421"]
print(solution(phone_book))
